Replicator/Replicator.cc

```diff
@@ -418,17 +406,17 @@ namespace litecore { namespace repl {
             logVerbose("Replicator status collection-wise: %s", statusVString().c_str());
         }
 
-        for (CollectionIndex i = 0; i < _collections.size(); ++i) {
-            if (_pullStatusV[i].error.code)
-                onError(_pullStatusV[i].error);
-            else if (_pushStatusV[i].error.code)
-                onError(_pushStatusV[i].error);
+        for (CollectionIndex i = 0; i < _subRepls.size(); ++i) {
+            if (_subRepls[i].pullStatus.error.code)
+                onError(_subRepls[i].pullStatus.error);
+            else if (_subRepls[i].pushStatus.error.code)
+                onError(_subRepls[i].pushStatus.error);
```

Comment:
What will happen when there are multiple onError() calls?

Reply:
I thought about it as well, but we may occur before the collections,
       if (_pullStatus.error.code)
            onError(_pullStatus.error);
        else if (_pushStatus.error.code)
            onError(_pushStatus.error);
  although not from different collection, but may come from puller and pusher. Now, it simply multiply by number of collections.
I guess, the first onError will throw if it is non-transient error.